### PR TITLE
Medication as needed

### PR DESCRIPTION
**Pull request by karenzheung**
_Pull requests into Flux require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable (**N/A**) and :white_check_mark:._

JIRA 949

Medication shows as needed for both visualizers. 

## Submitter:

**Running the Application**

- [x] Manually tested in Chrome
- [ ] Manually tested in IE

**Documentation**

- [x] This pull request describes why these changes were made
- [x] Recognizes any potential shortcomings/bugs in the description 
- [ ] Cheat sheet is updated
- [ ] Demo script is updated 
- [ ] Documentation on Wiki has been updated 
- [ ] Additional technical components and libraries are documented and added to wiki as needed

**NoteParser**

- [ ] Note parser has been updated if structured phrases change

**Code Quality**

- [x] 4-space indents - convert any tabs to spaces
- [x] Use public class field syntax for binding functions - ex. handleChange = () => { ... };
- [x] Code is commented

**Tests**

- [x] Existing tests passed
- [ ] Added Enzyme-UI tests for slim mode 
- [ ] Added Enzyme-UI tests for full mode
- [ ] Added backend tests for new functionality added
- [ ] Added note parser examples to test new functionality

## Reviewer 1: Name

- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

## Reviewer 2: Name

- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: src/summary/MedicationRangeChartVisualizer.jsx [hunk 1]
```diff
@@ -74,10 +74,11 @@ class MedicationRangeChartVisualizer extends Component {
         return rows;
     }
 
-    renderMedicationTitle = (lowerValue, upperValue, name, dosageValue, dosageUnit, timingValue, timingUnit) => {
+    renderMedicationTitle = (lowerValue, upperValue, name, dosageValue, dosageUnit, timingValue, timingUnit, asNeededIndicator) => {
         // Determining if timingUnit has a value. Set if empty string if null.
         if (timingUnit == null) timingUnit = '';
-
+        //Determining if asNeededIndicator is true. Set empty string if false, set string as needed if true.
+        asNeededIndicator ? asNeededIndicator = 'as needed' : asNeededIndicator = '';
```
Comment (Dtphelan1) [inline src/summary/MedicationRangeChartVisualizer.jsx]: [DP1] We probably shouldn't be changing argument values here as a best practice. I know it's just a string here, but it's probably not the best thing for us to have code-quality wise. Any chance we can just have a local variable here instead? Something like `const asNeededString = (asNeededIndicator ? "as needed" : "");`

**Comment by Dtphelan1** [comment]
@karenzheung when you get the chance if you could address my comments that be great :)

**Comment by karenzheung** [comment]
Fixed DP1

File: src/summary/TabularListVisualizer.jsx [hunk 2]
```diff
@@ -251,6 +251,10 @@ export default class TabularListVisualizer extends Component {
             const isInsertable = Lang.isUndefined(element.isInsertable) ? true : element.isInsertable;
             const isUnsigned = element.isUnsigned || false;
             let elementText = element.value;
+            if (arrayIndex === 3) {
```
Comment (gregquinn2001) [inline src/summary/TabularListVisualizer.jsx]: [GQ1] We can't change TabularListVisualizer to have code specific to any table. Incorporating asNeeded is specific to medications and tabular list visualizer should stay generic. I can explain better in person maybe...

**Comment by karenzheung** [comment]
Fixed GQ1

**Comment by Dtphelan1** [comment]
Just was waiting for DP2 to be addressed. All else looks good to me.

**Comment by karenzheung** [comment]
Fixed DP2!
